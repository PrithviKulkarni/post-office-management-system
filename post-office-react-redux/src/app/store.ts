import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
