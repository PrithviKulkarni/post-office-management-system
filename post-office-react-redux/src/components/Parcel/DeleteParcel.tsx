import { useNavigate, useParams } from 'react-router-dom'
import { useAppDispatch } from '../../app/hooks'
import { useSelector } from 'react-redux'
import { RootState } from '../../app/store'
import { getParcelById, deleteAParcel } from '../../features/ParcelSlice'
import { SyntheticEvent, useState } from 'react'
import { Parcel, defaultParcel } from '../interfaces/parcelInterface'
import { Container, Button, Form, InputGroup } from 'react-bootstrap'

/**
 *
 * @returns Looks into the individual parcel and deletes it
 */
const DeleteParcel = () => {
  const navigate = useNavigate()
  const dispatch = useAppDispatch()

  let { parcelId } = useParams()
  let parcel_Id: string = '0'
  if (!parcelId) {
    parcelId = '0'
  }
  parcel_Id = parcelId
  console.log({ parcel_Id })

  const parcels = useSelector((state: RootState) =>
    getParcelById(state, parcel_Id)
  )

  const [parcel_no, setParcelNo] = useState(parcels?.parcel_no)
  const [parcel_weight, setParcelW] = useState(parcels?.parcel_weight)
  const [status, setStatus] = useState(parcels?.status)
  const [estimated_delivery_days, setEDD] = useState(
    parcels?.estimated_delivery_days
  )
  const [errorMessage, setErrorMessage] = useState<string>('')

  const submitHandler = (event: SyntheticEvent) => {
    event.preventDefault()
    console.log('INSide the submit handler')
    try {
      dispatch(deleteAParcel(parcel_Id))
      navigate('/parcels')
      window.location.reload()
    } catch (error) {
      setErrorMessage('Parcel Not Found')
      console.log('errormessage: ', errorMessage)
    }
  }

  // if (parcel_Id === '0') {
  //   return (
  //     <>
  //       <section data-testid="deleteParcel" > </section>
  //     </>
  //   )
  // } else {
    return (
      <Container className="deleteParcel" data-testid="deleteParcel">
        <h4 className='text-center text-white'> Deleting the Parcel </h4>
        <Form onSubmit={submitHandler}   >
          <InputGroup className='mb-3'>
            <InputGroup.Text id='basic-addon1'>Parcel no. </InputGroup.Text>
            <Form.Control
              type='text'
              aria-label='parcel_weight'
              aria-describedby='basic-addon1'
              size='sm'
              name='parcel_no'
              value={parcel_no}
              onChange={event => setParcelNo(Number(event.target.value))}
            />
          </InputGroup>

          <InputGroup className='mb-3'>
            <InputGroup.Text id='basic-addon1'>Weight (kg)</InputGroup.Text>
            <Form.Control
              type='text'
              aria-label='parcel_weight'
              aria-describedby='basic-addon1'
              size='sm'
              name='parcel_weight'
              value={parcel_weight}
              onChange={event => setParcelW(Number(event.target.value))}
            />
          </InputGroup>

          <InputGroup className='mb-3'>
            <InputGroup.Text id='basic-addon1'>status</InputGroup.Text>
            <Form.Control
              type='text'
              aria-label='status'
              aria-describedby='basic-addon1'
              size='sm'
              name='status'
              value={status}
              onChange={event => setStatus(event.target.value)}
            />
          </InputGroup>

          <InputGroup className='mb-3'>
            <InputGroup.Text id='basic-addon1'>
              estimated_delivery_days
            </InputGroup.Text>
            <Form.Control
              type='text'
              aria-label='estimated_delivery_days'
              aria-describedby='basic-addon1'
              size='sm'
              name='estimated_delivery_days'
              value={estimated_delivery_days}
              onChange={event => setEDD(Number(event.target.value))}
            />
          </InputGroup>

          <Button type='submit' className='button' data-testid='parcelButton'>Delete Parcel Details</Button>
          <p>{errorMessage} </p>
        </Form>
      </Container>
    )
  }


export default DeleteParcel
