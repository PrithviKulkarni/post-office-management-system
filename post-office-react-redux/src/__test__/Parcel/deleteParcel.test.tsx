import React from "react";
import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import { store } from "../../app/store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import DeleteParcel from '../../components/Parcel/DeleteParcel';


afterEach(cleanup);
it("renders Delete Parcel page correctly", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <DeleteParcel />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Verifies the Delete Parcel render", () => {
  const {getByTestId } = render(
    <Provider store={store}>
      <BrowserRouter>
        <DeleteParcel />
      </BrowserRouter>
    </Provider>
  );


  const button = getByTestId('parcelButton')
  expect(button.textContent).toBe('Delete Parcel Details');
  fireEvent.click(button)
  expect(button).toBeTruthy()
  expect(screen.getByTestId("deleteParcel")).toBeInTheDocument()
});
