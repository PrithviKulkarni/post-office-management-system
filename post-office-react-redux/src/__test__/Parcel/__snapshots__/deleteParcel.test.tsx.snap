// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Delete Parcel page correctly 1`] = `
<DocumentFragment>
  <div
    class="deleteParcel container"
    data-testid="deleteParcel"
  >
    <h4
      class="text-center text-white"
    >
       Deleting the Parcel 
    </h4>
    <form
      class=""
    >
      <div
        class="mb-3 input-group"
      >
        <span
          class="input-group-text"
          id="basic-addon1"
        >
          Parcel no. 
        </span>
        <input
          aria-describedby="basic-addon1"
          aria-label="parcel_weight"
          class="form-control form-control-sm"
          name="parcel_no"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-3 input-group"
      >
        <span
          class="input-group-text"
          id="basic-addon1"
        >
          Weight (kg)
        </span>
        <input
          aria-describedby="basic-addon1"
          aria-label="parcel_weight"
          class="form-control form-control-sm"
          name="parcel_weight"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-3 input-group"
      >
        <span
          class="input-group-text"
          id="basic-addon1"
        >
          status
        </span>
        <input
          aria-describedby="basic-addon1"
          aria-label="status"
          class="form-control form-control-sm"
          name="status"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-3 input-group"
      >
        <span
          class="input-group-text"
          id="basic-addon1"
        >
          estimated_delivery_days
        </span>
        <input
          aria-describedby="basic-addon1"
          aria-label="estimated_delivery_days"
          class="form-control form-control-sm"
          name="estimated_delivery_days"
          type="text"
          value=""
        />
      </div>
      <button
        class="button btn btn-primary"
        data-testid="parcelButton"
        type="submit"
      >
        Delete Parcel Details
      </button>
      <p>
         
      </p>
    </form>
  </div>
</DocumentFragment>
`;
